libraries/AP_NavEKF3/AP_NavEKF3.cpp

```diff
@@ -1143,6 +1144,34 @@ bool NavEKF3::pre_arm_check(bool requires_position, char *failure_msg, uint8_t f
         return false;
     }
 
+    // check if range finder is enabled and in range
+#if AP_RANGEFINDER_ENABLED
+    if (requires_position || requires_height) {
+        const AP_NavEKF_Source::SourceZ height_source = sources.getPosZSource();
+        if (height_source == AP_NavEKF_Source::SourceZ::RANGEFINDER) {
+            const auto *_rng = dal.rangefinder();
+            if (_rng == nullptr) {
+                dal.snprintf(failure_msg, failure_msg_len, "Rangefinder not enabled");
+                return false;
+            }
+            bool rangefinder_healthy = false;
```

Comment:
can't we just check rngValidMeaTime_ms ?

Reply:
should be in the new pre_arm check in the backends, so we check all lanes, check it is fusing